pkg/k8s/resource.go

```diff
@@ -137,6 +137,27 @@ func (r Resource) Kind() string {
 	}
 }
 
+// QKind returns a fully qualified resource kind with the following
+// format: <kind>.<version>.<group>
+func (r Resource) QKind() string {
+	gv, ok := r["apiVersion"]
+	if ok {
+		if strings.ToLower(gv.(string)) == "v1" {
+			return r.Kind()
```

Comment:
Why Does this just return `Kind` instead of `Kind.v1` or even `Kind.v1.core`?  IMO, the right thing to do if `gv.(string)` doesn't contain a slash is to set `version=gv.(string)` and `group="core"`.

Reply:
Dammit, this is just something that Kubernetes is inconsistent about.

The proper QKind for a `Service` is `Service.v1.`, so `group=""`, not `group="core"`.

Reply:
The first version of the code produced `Service.v1.`, but the resolution code doesn't recognize that. You end up with "no such resource: Service" coming out of the kubernetes machinery somewhere. The docs here: https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups specify that the v1 version is a special legacy case, and given your point in the ResolveResourceType comment about some versions using "" and some versions using "core", I wasn't really sure of a safe way to qualify it, so I opted to just use the naked kind for the legacy case. Did I misunderstand the comment, is there something better to do here?

Reply:
> but the resolution code doesn't recognize that.

Are you sure about that?
```console
$ kubectl get Service
NAME                  TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
consul                ClusterIP   10.99.91.214    <none>        8500/TCP   1h
kubernetes            ClusterIP   10.96.0.1       <none>        443/TCP    7h
teleproxied-httpbin   ClusterIP   10.105.54.188   <none>        80/TCP     1h

$ kubectl get Service.
NAME                  TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
consul                ClusterIP   10.99.91.214    <none>        8500/TCP   1h
kubernetes            ClusterIP   10.96.0.1       <none>        443/TCP    7h
teleproxied-httpbin   ClusterIP   10.105.54.188   <none>        80/TCP     1h

$ kubectl get Service.v1
error: the server doesn't have a resource type "Service"

$ kubectl get Service.v1.
NAME                  TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
consul                ClusterIP   10.99.91.214    <none>        8500/TCP   1h
kubernetes            ClusterIP   10.96.0.1       <none>        443/TCP    7h
teleproxied-httpbin   ClusterIP   10.105.54.188   <none>        80/TCP     1h

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"archive", BuildDate:"2019-06-20T22:55:20Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:34:22Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}```

Reply:
I guess not. Maybe the last dot was actually punctuation from the error message.